src/Microsoft.DotNet.XHarness.iOS.Shared/Hardware/SimulatorLoader.cs

```diff
@@ -462,6 +462,31 @@ public ISimulatorDevice FindCompanionDevice(ILog log, ISimulatorDevice device)
             return _availableDevices.Single(v => v.UDID == pair.Companion);
         }
 
+        public async Task<(ISimulatorDevice Simulator, ISimulatorDevice? CompanionSimulator)> FindSimulators(TestTargetOs target, ILog log, int retryCount, bool createIfNeeded = true, bool minVersion = false)
+        {
+            int attempt = 1;
+            while (true)
+            {
+                try
+                {
+                    return await FindSimulators(target, log);
+                }
+                catch (Exception e)
+                {
+                    log.WriteLine($"Failed to find/create simulator (attempt {attempt}/{retryCount}):" + Environment.NewLine + e);
+
+                    if (attempt == retryCount)
```

Comment:
Could `retryCount` be 0?

Reply:
Good catch

Reply:
Sorry, did you add the change to handle that case? I cannot find it

Reply:
Oops, committed to a wrong branch! Thanks for noticing